Riot/Managers/KeyValueStorage/KeychainStore.swift

```diff
@@ -35,14 +52,23 @@ class KeychainStore: KeyValueStore {
             return
         }
         
-        if let value = value as? String {
+        if let value = value as? Data {
             try keychain.set(value, key: key)
-        } else if let value = value as? Data {
+        } else if let value = value as? String {
+            try keychain.set(value, key: key)
+        } else if let value = value as? Bool {
             try keychain.set(value, key: key)
         }
     }
 
     func object(forKey key: KeyValueStoreKey) throws -> Any? {
```

Comment:
As we want to know the output type. We can have an API close to `KeychainAccess` or `UserDefault` and have getter for the 3 supported types. It will avoid this routing.